### PR TITLE
Aida 657

### PR DESCRIPTION
**Pull request by ecurley-nextcentury**
Implement temporal properties

### CODE REVIEW

File: src/main/java/com/ncc/aif/AIFUtils.java [hunk 1]
```diff
@@ -1117,11 +1121,67 @@ public static Resource linkToExternalKB(Model model, Resource toLink, String ext
...
+        private final LDCTimeType type;
+        private final String year;
+        private final String month;
+        private final String date;
+
+        public LDCTimeComponent(LDCTimeType type, String year, String month, String date) {
+            this.type = type;
+            this.year = year;
+            this.month = month;
+            this.date = date;
```
Comment (sharknc) [inline src/main/java/com/ncc/aif/AIFUtils.java]: I would update the name of the `date` parameter to `day` so it is not confusing that this field actually represents a day.

File: src/main/java/com/ncc/aif/AIFUtils.java [hunk 2]
```diff
@@ -1117,11 +1121,67 @@ public static Resource linkToExternalKB(Model model, Resource toLink, String ext
...
+        }
+
+        private Resource makeAIFTimeComponent(Model model) {
+            final Resource timeComponent = makeAIFResource(model, null, AidaAnnotationOntology.LDC_TIME_COMPONENT, null);
+            timeComponent.addProperty(AidaAnnotationOntology.LDC_TIME_TYPE, type.toString());
+            addInteger(model, timeComponent, AidaAnnotationOntology.LDC_TIME_YEAR, year, XSD.gYear);
+            addInteger(model, timeComponent, AidaAnnotationOntology.LDC_TIME_MONTH, month, XSD.gMonth);
+            addInteger(model, timeComponent, AidaAnnotationOntology.LDC_TIME_DATE, date, XSD.gDay);
```
Comment (sharknc) [inline src/main/java/com/ncc/aif/AIFUtils.java]: Same as above, I recommend updating the `LDC_TIME_DATE` to be `LDC_TIME_DAY`.

File: src/main/java/com/ncc/aif/AidaAnnotationOntology.java [hunk 3]
```diff
@@ -63,6 +63,20 @@
...
+            ResourceFactory.createProperty(NAMESPACE + "start");
+    public static final Property LDC_TIME_END =
+            ResourceFactory.createProperty(NAMESPACE + "end");
+    public static final Property LDC_TIME_TYPE =
+            ResourceFactory.createProperty(NAMESPACE + "timeType");
+    public static final Property LDC_TIME_YEAR =
+            ResourceFactory.createProperty(NAMESPACE + "year");
+    public static final Property LDC_TIME_MONTH =
+            ResourceFactory.createProperty(NAMESPACE + "month");
+    public static final Property LDC_TIME_DATE =
```
Comment (sharknc) [inline src/main/java/com/ncc/aif/AidaAnnotationOntology.java]: If you agree with the comments above, update `LDC_TIME_DATE` to `LDC_TIME_DAY` and update the property to "day".

File: src/main/resources/com/ncc/aif/aida_ontology.shacl [hunk 4]
```diff
@@ -863,6 +815,69 @@ aida:JustificationPropertyShape
...
+    sh:property [
+        sh:path aida:year ;
+        sh:datatype xsd:gYear
+    ] ;
+    sh:property [
+        sh:path aida:month ;
+        sh:datatype xsd:gMonth
+    ] ;
+    sh:property [
+        sh:path aida:date ;
```
Comment (sharknc) [inline src/main/resources/com/ncc/aif/aida_ontology.shacl]: If you agree with the comments above, update `aida:date` to `aida:day`

File: src/main/resources/com/ncc/aif/aida_ontology.shacl [hunk 5]
```diff
@@ -109,13 +109,13 @@ aida:EntityShape
 # (e.g. an attack, a marriage, a business acquisition)
 # Each event must have at least one assertion giving it a type which is a sub-class of
 # aidaDomainCommon:eventType
-aida:EventShape
+aida:EventRelationShape
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply src/main/resources/com/ncc/aif/aida_ontology.shacl]: Yes, but as we identify which node is the actual problem, it should be relatively easy to figure out. Even if they diverge later, they're the same shape now and we should treat them as such. Let me know if you disagree

**Pull request merged**